packages/zowe-explorer/src/configuration/Profiles.ts

```diff
@@ -97,37 +98,67 @@ export class Profiles extends ProfilesCache {
         void Gui.errorMessage(inactiveMsg);
     }
 
+    /**
+     * Checks if the profile has a secure token.
+     * Note: This is a workaround to maintain backward compatibility.
+     * @param theProfile - The profile to check.
+     * @returns True if the profile has a secure token, false otherwise.
+     */
+    private async profileHasSecureToken(theProfile: imperative.IProfileLoaded): Promise<boolean> {
+        const teamConfig = (await this.getProfileInfo()).getTeamConfig();
+        const profName = teamConfig.api.profiles.getProfilePathFromName(theProfile.name);
+        const tokenValue = profName + ".properties.tokenValue";
+        return teamConfig.api.secure.secureFields().includes(tokenValue);
+    }
+
     public async checkCurrentProfile(theProfile: imperative.IProfileLoaded, node?: Types.IZoweNodeType): Promise<Validation.IValidationProfile> {
```

Comment:
It seems like `node` is now passed to this function in various places. Was this change required to keep the same functionality as before? Or are we performing additional logic now that requires the node for full effect?